subsys/net/ip/Kconfig.tcp

```diff
@@ -244,4 +244,13 @@ config NET_TCP_REJECT_CONN_WITH_RST
 	  If enabled, TCP stack will reject connection attempts on unbound ports
 	  with TCP RST packet.
 
+config NET_TCP_IPV6_ND_REACHABILITY_HINT
+	bool "Provide a reachability hint for IPv6 Neighbor Discovery"
+	depends on NET_TCP
+	depends on NET_IPV6_ND
+	help
+	  If enabled, TCP stack will inform the IPv6 Neighbor Discovery process
+	  about the active link to a specific neighbor by signaling recent
+	  "forward progress" event as described in RFC 4861.
```

Comment:
Shouldn't we enable this by default, looks like a very useful enhancement, @rlubos WDYT?

Reply:
Perhaps that'd indeed make sense, yes.

Reply:
@LuDuda could you enable the option by default, it is quite small (in memory usage) but avoids extra network traffic packets which is very useful.

Reply:
I've enabled this feature by default. However, I'm wondering if the feature is useful for situations where we're fighting for every fraction of TCP performance - where searching in a neighbor table might be considered an overhead. Perhaps, for such cases, one could simply disable this feature, or I could implement a simple rate-limiting mechanism by recording the time of the last received ACK in `struct tcp` - but that will increase the overall memory size. What do you think?

Reply:
> Perhaps, for such cases, one could simply disable this feature, or I could implement a simple rate-limiting mechanism by recording the time of the last received ACK in `struct tcp` - but that will increase the overall memory size.

Sure, we could experiment with the rate limiting. It would actually be nice to get information is there a performance hit by this PR, we could run zperf to get that info. We could still adjust the default value if there is performance problems with this change.